yt_dlp/networking/common.py

```diff
@@ -263,32 +270,22 @@ def _check_proxies(self, proxies):
             if scheme not in self._SUPPORTED_PROXY_SCHEMES:
                 raise UnsupportedRequest(f'Unsupported proxy type: "{scheme}"')
 
-    def _check_cookiejar_extension(self, extensions):
-        if not extensions.get('cookiejar'):
-            return
-        if not isinstance(extensions['cookiejar'], CookieJar):
-            raise UnsupportedRequest('cookiejar is not a CookieJar')
-
-    def _check_timeout_extension(self, extensions):
-        if extensions.get('timeout') is None:
-            return
-        if not isinstance(extensions['timeout'], (float, int)):
-            raise UnsupportedRequest('timeout is not a float or int')
-
     def _check_extensions(self, extensions):
-        self._check_cookiejar_extension(extensions)
-        self._check_timeout_extension(extensions)
-
-    def _validate(self, request):
-        self._check_url_scheme(request)
-        self._check_proxies(request.proxies or self.proxies)
-        self._check_extensions(request.extensions)
+        """Check extensions for unsupported extensions. Subclasses should extend this."""
+        assert isinstance(extensions.get('cookiejar'), (CookieJar, types.NoneType))
+        assert isinstance(extensions.get('timeout'), (float, int, types.NoneType))
 
     @wrap_request_errors
     def validate(self, request: Request):
         if not isinstance(request, Request):
             raise TypeError('Expected an instance of Request')
-        self._validate(request)
```

Comment:
We need `_validate_` cause we don't want people overriding `validate()`, since we need `wrap_request_errors` to apply.